### PR TITLE
Fix failing macOS build.

### LINKED ISSUE #12657: Mac OS X CI builds fail at installing ruby
```
==> Pouring adwaita-icon-theme-3.32.0.mojave.bottle.tar.gz
🍺  /usr/local/Cellar/adwaita-icon-theme/3.32.0: 5,827 files, 22.6MB
Error: /usr/local/opt/ruby@2.6 is not a valid keg
```
https://github.com/coq/coq/pull/12556/checks?check_run_id=847065961

I also reported this at https://github.com/actions/virtual-environments/issues/1198, though it's possible this it just an issue with Coq's build script rather than the OSX build environment (maybe the homebrew commit is too old)?

### PR DESCRIPTION
**Pull request by Zimmi48**
This works around #12657.

### CODE REVIEW

**Comment by ejgallego** [comment]
Umm, the bump altered somehow the gtk install and the dmg creation doesn't work now. We need help from a OSX user.

**Comment by Zimmi48** [comment]
Not sure an OSX user would be sufficient to debug this. @maximedenes is the one who authored the build script and probably the only one who understands what it does...

**Comment by Zimmi48** [comment]
Another (temporary) solution would to bump to a less recent version that does not contain the latest gtk version.

**Comment by Zimmi48** [comment]
OK, let's do like this.

File: azure-pipelines.yml [hunk 1]
```diff
@@ -64,7 +64,7 @@ jobs:
       set -e
       brew update
       (cd $(brew --repository)/Library/Taps/homebrew/homebrew-core/ && git fetch --shallow-since=${HBCORE_DATE} && git checkout ${HBCORE_REF})
-      brew install gnu-time opam pkg-config gtksourceview3 adwaita-icon-theme
+      brew install gnu-time opam pkg-config gtksourceview3 adwaita-icon-theme || true
```
Comment (JasonGross) [inline azure-pipelines.yml:67]: If you're going to ignore the failure, why not just remove the failing package?
```suggestion
      brew install gnu-time opam pkg-config gtksourceview3 # adwaita-icon-theme
```

Comment (Zimmi48) [reply azure-pipelines.yml:67]: Is this the failing package? From the discussion on Zulip, this was not my understanding.

**Comment by ejgallego** [comment]
Well it works but likely there won't be icons in CoqIDE; I had understood that even if ruby failed the icon theme would be installed. Still I don't see why ruby is pulled tho. Maybe `hicolor-icon-theme` is a good alternative?

**Comment by Zimmi48** [comment]
So it seems that Jason is right: the failure happens during the installation of `adwaita-icon-theme`, and it's an issue that while just ignoring the error, we won't be fully sure what went through and what didn't.

If we're to change for another icon set, people might be surprised of the change in icons that isn't really justified by anything.

Have we tried the other suggestion from @Blaisorblade which was to pin brew itself to an old version?

**Comment by Blaisorblade** [comment]
Can you state the evidence that the icons are not installed? The build seems to even copy them successfully —

https://dev.azure.com/coq/coq/_build/results?buildId=11008&view=logs&jobId=a5e52b91-c83f-5429-4a68-c246fc63a4f7&j=a5e52b91-c83f-5429-4a68-c246fc63a4f7&t=82eb5711-24fa-5d76-924d-28dd4736936e

> we won't be fully sure what went through and what didn't.

I agree: I’m aware my suggestion is sinful, but I suggested that as a stop-gap, because it seemed better than delaying the 8.12 release or porting CoqIDE to a new Gtk (which would also delay the release).

One could run `brew list` (IIRC, I can check) to confirm what packages are installed, and test the produced package (you can send me a link).

> I had understood that even if ruby failed the icon theme would be installed.

That’s still my best guess from the given evidence. The log was that the error happened after the icon theme was poured.

> Still I don't see why ruby is pulled tho

Ruby is not a actual dependency of the icon set, but lots of hints in Brew suggests it uses ruby 2.6 itself. I’m not sure why it happens then.

**Comment by ejgallego** [comment]
> Have we tried the other suggestion from @Blaisorblade which was to pin brew itself to an old version?

I have no idea how to do this, I'm afraid.

**Comment by ejgallego** [comment]
> Can you state the evidence that the icons are not installed? The build seems to even copy them successfully —

We had this bug that if that package was not installed, CoqIDE would miss some icons.

You can test the artifact here https://dev.azure.com/coq/coq/_build/results?buildId=11008&view=artifacts&type=publishedArtifacts

**Comment by ejgallego** [comment]
I'm gonna push a different hack based on Théo's first attempt to update more.

**Comment by ejgallego** [comment]
So indeed the problem is that recent brew changed how it handles the `dylib` files for GTK, I'm trying to see if we can ignore this and still get a working GTK install.

**Comment by gares** [comment]
I'm getting bombed by notifications about commits that are really not meant to be reviewed, I think you should go back to draft mode.

**Comment by Zimmi48** [comment]
> You can test the artifact here https://dev.azure.com/coq/coq/_build/results?buildId=11008&view=artifacts&type=publishedArtifacts

@Blaisorblade Did you get a chance to test this installer?
Maybe we can ask some other macOS user as well, like @anton-trunov.

We're probably going to release very soon, so either we go with this (previous) fix (provided that the generated CoqIDE has its icons or is still reasonable) or we will release without a macOS installer.

**Comment by Blaisorblade** [comment]
Here it is, and the icons are there! I assume it'll work on 8.12 as well.
image

image

**Comment by Zimmi48** [comment]
Let's commit this temporary fix then!

File: azure-pipelines.yml [hunk 2]
```diff
@@ -64,7 +64,8 @@ jobs:
...
+      # || true: workaround #12657, see also #12672 and commit message for this line
```
Comment (SkySkimmer) [inline azure-pipelines.yml:68]: I just added this comment.

**Comment by Blaisorblade** [comment]
And thanks for the ping, I had already lost this notification. Maybe you could have a mac-users or mac-reviewers group to tag in these cases?

**Review by SkySkimmer — APPROVED** [review]
No point waiting, I'll merge now.

**Pull request merged**